### PR TITLE
Add flag to indicate if ROMclass is from SCC

### PR DESCRIPTION
**Pull request by theresa-m**
Taking over from https://github.com/eclipse-openj9/openj9/pull/19637

Documentation  update: https://github.com/eclipse-openj9/openj9-docs/issues/1406

### CODE REVIEW

File: runtime/bcutil/ROMClassBuilder.cpp [hunk 1]
```diff
@@ -768,10 +768,13 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
 			 */
 			if (!sharedStoreClassTransaction.isCacheFull()) {
 				if ( sharedStoreClassTransaction.allocateSharedClass(&sizeRequirements) ){
-					U_8 *romClassBuffer = (U_8*)sharedStoreClassTransaction.getRomClass();
+					J9ROMClass *romClassBuffer = (J9ROMClass*)sharedStoreClassTransaction.getRomClass();
 					/*
 					 * Make note that the laydown is occurring in SharedClasses
 					 */
+
+					romClassBuffer->extraModifiers |= J9AccClassIsStoredInSCC;
```
Comment (hangshao0) [inline runtime/bcutil/ROMClassBuilder.cpp:776]: This won't work. This should be `extraModifiers |= J9AccClassIsStoredInSCC`.

File: runtime/oti/j9javaaccessflags.h [hunk 2]
```diff
@@ -99,6 +99,7 @@
  * See map in ROMClassBuilder::computeExtraModifiers for
  * available slots.
  */
+#define J9AccClassIsStoredInSCC 0x20
```
Comment (hangshao0) [inline runtime/oti/j9javaaccessflags.h:102]: `J9AccClassIsShared` is a better name.

**Comment by theresa-m** [comment]
Thanks @hangshao0 I opened this documentation issue https://github.com/eclipse-openj9/openj9-docs/issues/1406

**Comment by hangshao0** [comment]
Jenkins test sanity,extended amac jdk21

**Comment by hangshao0** [comment]
Jenkins test sanity,extended zlinux jdk8

**Comment by hangshao0** [comment]
PR builds (all passed): 
https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/6034/
https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/6035/
https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/6036/

File: runtime/oti/j9javaaccessflags.h [hunk 3]
```diff
@@ -99,6 +99,7 @@
  * See map in ROMClassBuilder::computeExtraModifiers for
  * available slots.
  */
+#define J9AccClassIsShared 0x20
```
Comment (hangshao0) [inline runtime/oti/j9javaaccessflags.h:102]: Please add the new flag into the comment here:

https://github.com/eclipse-openj9/openj9/blob/0c6cdfd79fe3f612c6bb80ba72bd9a02bfa13ca7/runtime/bcutil/ROMClassBuilder.cpp#L1219

Comment (theresa-m) [reply runtime/oti/j9javaaccessflags.h:102]: I updated the comment.

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**